scripts/tests/run_tv_casting_test.py

```diff
@@ -0,0 +1,245 @@
+def validate_value(expected_value, line, value_name):
+
+    # Extract the integer value from the string.
+    value = extract_value_from_string(line)
+
+    # If the discovered value does not match the expected value,
+    # print the error and return False.
+    if value != expected_value:
+        logging.error(f'{value_name} does not match the expected value!')
+        logging.error(f'Expected {value_name}: {expected_value}')
+        line = line.rstrip('\n')
+        logging.error(line)
+
+        return False
+
+    # Return True if the value matches the expected value.
+    return True
+
+
+# Tear down the processes once done running.
+def tear_down(processes):
+
+    for process in processes:
+        process.terminate()
+        process.wait()
+
+
+# Test if the Linux tv-casting-app is able to discover the Linux tv-app.
+# The Linux tv-casting-app and the tv-app will be run in separate processes.
+# Their corresponding output will be written to their respective log files.
+# The output of the tv-casting-app will be parsed in realtime for strings of
+# interest which will be printed to the console.
+def test_discovery_fn():
+
+    with open(LINUX_TV_APP_LOGS, 'w') as fd1, open(LINUX_TV_CASTING_APP_LOGS, 'w') as fd2:
+
+        # Run the Linux tv-app and write the output to file.
+        tv_app_rel_path = 'out/tv-app/chip-tv-app'
+        tv_app_abs_path = os.path.abspath(tv_app_rel_path)
+        tv_app_process = subprocess.Popen(tv_app_abs_path, stdout=fd1, stderr=subprocess.PIPE, text=True)
+
+        time.sleep(RUN_INTERVAL)
+
+        # Run the Linux tv-casting-app.
+        tv_casting_app_rel_path = 'out/tv-casting-app/chip-tv-casting-app'
+        tv_casting_app_abs_path = os.path.abspath(tv_casting_app_rel_path)
+        tv_casting_app_process = subprocess.Popen(
+            tv_casting_app_abs_path, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+        # Initialize variables.
+        continue_parsing = False
+        valid_discovered_commissioner_str = ''
+
+        # Read the output as we get it from the tv-casting-app process.
+        for line in tv_casting_app_process.stdout:
+
+            # Write the line to the Linux tv-casting-app log file.
+            fd2.write(line)
+
+            # Fail fast if "No commissioner discovered" string found.
+            if "No commissioner discovered" in line:
+
+                line = line.rstrip('\n')
+                logging.error(line)
+                handle_discovery_failure()
+
+            # Look for 'Discovered Commissioner'.
+            if "Discovered Commissioner" in line:
+
+                line = line.rstrip('\n')
+                valid_discovered_commissioner_str = line
+
+                # Continue parsing the content that belongs to the "Discovered Commissioner".
+                continue_parsing = True
+
+                # Initialize variables to store the information of interest.
+                valid_vendor_id = False
```

Comment:
Make the values as `Option[str]` and then None means "no value yet" and a valid string means a value. then you do not need both bool and str.

Reply:
This is addressed in the latest commit.